entity-management-web/src/main/java/eu/europeana/entitymanagement/batch/config/EntityUpdateJobFactory.java

```diff
@@ -0,0 +1,149 @@
+                .start(synchronousUpdate(processors, writers))
+                .build();
+    }
+
+    /**
+     * Creates a step for synchronous update
+     * @return
+     */
+    private Step synchronousUpdate(List<Task> processors, List<Task> writers) {
+        return this.stepBuilderFactory
+                .get(STEP_UPDATE_ENTITY)
+                .<BatchEntityRecord, BatchEntityRecord>chunk(1)
+                .reader(getReader(true))
+                .processor(getProcessor(processors))
+                .writer(getWriter(writers))
+                .listener((ItemProcessListener<? super BatchEntityRecord, ? super BatchEntityRecord>) itemListener)
+                .faultTolerant()
+                .skipPolicy(noopSkipPolicy)
+                .taskExecutor((TaskExecutor)getApplicationContext().getBean(WEB_REQUEST_JOB_EXECUTOR))
+                .throttleLimit(emConfig.getBatchUpdatesThrottleLimit())
+                .listener(stepExecutionListener(List.of(ScheduledUpdateType.FULL_UPDATE), true))
+                .build();
+
+    }
+
+    /**
+     * Creates Item reader based on synchronus and async updates
+     * @param synchronous
+     * @return
+     */
+    private ItemReader<BatchEntityRecord> getReader(boolean synchronous) {
+        return synchronous ? (EntityRecordDatabaseReader) getApplicationContext().getBean(SINGLE_ENTITY_RECORD_READER)
+                : (SynchronizedItemStreamReader)getApplicationContext().getBean(SCHEDULED_TASK_READER);
+    }
+
+    private ItemProcessor<BatchEntityRecord, BatchEntityRecord> getProcessor(List<Task> processors) {
+        return isFullUpdate(processors) ?
+                (ItemProcessor<BatchEntityRecord, BatchEntityRecord>) getApplicationContext().getBean(FULL_ENTITY_UPDATE_PROCESSOR)
+                 : emAutoConfig.compositeProcessor(processors);
+
+    }
+
+    private ItemWriter<BatchEntityRecord> getWriter(List<Task> writers) {
+        ItemWriter<BatchEntityRecord> writer = new CompositeItemWriter<>();
+        if (writers.contains(Task.DB_UPDATE) && writers.contains(Task.SOLR_INSERTION)) {
+            writer = (ItemWriter<BatchEntityRecord>) getApplicationContext().getBean(ENTITY_UPDATE_WRITERS);
+        } else if (writers.contains(Task.DB_UPDATE)) {
+            writer = emAutoConfig.recordDBInsertionWriter();
+        } else if (writers.contains(Task.SOLR_INSERTION)) {
+            writer = emAutoConfig.entitySolrInsertionWriter();
+        }
+        return writer;
+    }
+
+    /** Creates a StepExecutionListener that's called before / after the step runs
+     *
+     * */
+    private StepExecutionListener stepExecutionListener(
+            List<? extends ScheduledTaskType> updateType, boolean isSynchronous) {
+        return new EntityUpdateStepListener(
+                scheduledTaskService, updateType, isSynchronous, emConfig.getMaxFailedTaskRetries());
+    }
+
+    private boolean isFullUpdate(List<Task> processors) {
```

Comment:
instead of using this method, better have a static List with the names of the processors. The implementation should not guess if it is a full update or not, that is infromation that is known in the API methods, should only be propertly passed to the subsequent method. Actually we could create a Job description object to keep the list of the processors, the list of writers and the type of the update (string).